### PR TITLE
Fixing pyvisitor template so that no node copies are done

### PR DESCRIPTION
**Pull request by ferdonline**
This will fix the Python visitor modifier which didn't modify the underlying AST, only a copy.

Tech detail:
We need to provide Pybind11 a reference wrapper object so it understands it's a reference and not a new object.

### CODE REVIEW

File: src/language/templates/pybind/pyvisitor.cpp [hunk 1]
```diff
@@ -103,25 +103,25 @@ namespace py = pybind11;
 
 {% for node in nodes %}
 void PyVisitor::visit_{{ node.class_name|snake_case }}(ast::{{ node.class_name }}& node) {
-    PYBIND11_OVERLOAD_PURE(void, Visitor, visit_{{ node.class_name|snake_case }}, node);
+    PYBIND11_OVERLOAD_PURE(void, Visitor, visit_{{ node.class_name|snake_case }}, std::ref(node));
```
Comment (pramodk) [inline src/language/templates/pybind/pyvisitor.cpp:106]: Someone needs to know all the secrete / hidden documentation of pybind11 🙄 🎉

Comment (ferdonline) [reply src/language/templates/pybind/pyvisitor.cpp:106]: hidden or non-existing... this was after a lot of digging :/

Comment (pramodk) [reply src/language/templates/pybind/pyvisitor.cpp:106]: Ok...Worth an issue or PR in the upstream documentation then!

Comment (ohm314) [reply src/language/templates/pybind/pyvisitor.cpp:106]: oh wow.. i did not even know `std::ref` . amazing!

File: test/unit/pybind/test_visitor.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 import nmodl
 from nmodl.dsl import ast, visitor
 import pytest
+import sys
```
Comment (ohm314) [inline test/unit/pybind/test_visitor.py:11]: not needed

**Review by ohm314 — APPROVED** [review]
Fantastic! This fixes #702

File: src/language/templates/pybind/pyvisitor.cpp (hunk 1 above)
Comment (cattabiani) [reply src/language/templates/pybind/pyvisitor.cpp:106]: +1 to Omar's comment

**Review by pramodk — APPROVED** [review]

**Pull request merged**